osp-deploy/cinder/powerflex/README.md

```diff
@@ -152,3 +156,69 @@ Install the Storage Data Client (SDC) on all nodes after deploying the overcloud
+
+### Upgrading RHOSP
+
+For instructions of how to upgrade RHOSP13 to RHOSP16.1 please refer to [In-place upgrades from Red Hat OpenStack Platform 13 to 16.1](https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/16.1/html/framework_for_upgrades_13_to_16.1).
+
+### Additional steps that have to be done to get working environment after upgrade
+
+If you use the environment file with configuration from previous installation (RHOSP13) some changes have to be done:
+
+* No custom containers for cinder/nova/glance services are needed in RHOSP16.1. They must be removed from environment file.
+  However, custom volume mappings are still necessary.
+
+* **cinder.volume.drivers.dell_emc.scaleio.driver.ScaleIODriver** was deprecated and removed. sio_* options have to be renamed.
+  Do not change **volume_backend_name** as it is used in Cinder Volume Type object.
+
+This file has to be included in openstack `overcloud upgrade prepare` and `openstack overcloud upgrade converge` commands.
+
+Example of configuration file changes:
+
+```diff
+parameter_defaults:  
+- DockerCinderVolumeImage: 192.168.24.1:8787/dellemc/openstack-cinder-volume-dellemc-vxflexos
+- DockerCinderBackupImage: 192.168.24.1:8787/dellemc/openstack-cinder-volume-dellemc-vxflexos
+- DockerNovaComputeImage: 192.168.24.1:8787/dellemc/openstack-nova-compute-dellemc-vxflexos
+- DockerGlanceApiImage: 192.168.24.1:8787/dellemc/openstack-glance-api-dellemc-vxflexos
+- DockerGlanceApiConfigImage: 192.168.24.1:8787/dellemc/openstack-glance-api-dellemc-vxflexos
+- DockerInsecureRegistryAddress:
+-   - 192.168.24.1:8787
+  NovaComputeOptVolumes:
+    - /opt/emc/scaleio/openstack:/opt/emc/scaleio/openstack
+  CinderVolumeOptVolumes:
+    - /opt/emc/scaleio/openstack:/opt/emc/scaleio/openstack
+  GlanceApiOptVolumes:
+    - /opt/emc/scaleio/openstack:/opt/emc/scaleio/openstack
+  GlanceBackend: cinder
+  ControllerExtraConfig:
+    cinder::config::cinder_config:
+      scaleio/volume_driver:
+-       value: cinder.volume.drivers.dell_emc.scaleio.driver.ScaleIODriver
++       value: cinder.volume.drivers.dell_emc.powerflex.driver.PowerFlexDriver
+      scaleio/volume_backend_name:
+        value: scaleio
+      scaleio/san_ip:
+        value: <PowerFlex GATEWAY IP>
+      scaleio/san_login:
+        value: <PowerFlex_USER>
+      scaleio/san_password:
+        value: <PowerFlex_PASSWD>
+-     scaleio/sio_storage_pools:
++     scaleio/powerflex_storage_pools:
+        value: <Comma-separated list of protection domain:storage pool name>
+-     sio_allow_non_padded_volumes:
++     powerflex_allow_non_padded_volumes:
+        value: True
+    cinder_user_enabled_backends: ['scaleio']
+```
+
+**NOTICE**: The OS of the Overcloud node is upgraded by issuing `openstack overcloud upgrade run --stack <STACK NAME> --tags system_upgrade --limit <NODE TO UPGRADE>`. 
+After OS upgrade of each Overcloud node where PowerFlex SDC is installed SDC package has to be reinstalled using
```

Comment:
why? Are you saying that powerflex 3.5.1.1 that runs on RHOSP 13 RHEL-7.9 will need to be reinstalled for the same powerflex version but for RHEL 8.2?
What about MDS and SDS and gateway if they run on overcloud nodes?

Reply:
At least SDC for RHEL7.9 won't start on RHEL8.2 as it contains kernel module.

Reply:
Should we add command here? or refer back to powerflex upgrade doc for it?

Reply:
Hopefully all admins who upgrade RHOSP know how to install/uninstall RPMs.

Reply:
ok